src/test/java/BQJDBC/QueryResultTest/JdbcUrlTest.java

```diff
@@ -120,6 +122,26 @@ public void canConnectWithJsonAuthFileContentsInProperties() throws SQLException
         stmt.executeQuery("SELECT * FROM orders limit 1");
     }
 
+    @Test
+    public void canConnectWithOAuthAccessToken() throws SQLException, IOException, GeneralSecurityException {
+        // generate access token from service account credentials
+        Properties serviceProps = getProperties("/protectedaccount.properties");
+        String accessToken = Oauth2Bigquery.generateAccessToken(
+            serviceProps.getProperty("user"),
+            serviceProps.getProperty("path"),
+            serviceProps.getProperty("password"),
+            null
+        );
+
+        Properties oauthProps = getProperties("/oauthaccount.properties");
+        oauthProps.setProperty("password", accessToken);
```

Comment:
oh so this is just our properties file, which is parsed and turned into a url here:

https://github.com/jonathanswenson/starschema-bigquery-jdbc/pull/41/files#diff-a1fea11022c0183fba5d1083384f962dR110-R120

I just made it password to not add a new property to parse, but I could also change it to be called oauthaccesstoken since that probably is clearer 🤷‍♀